### DISCUSSION TITLE
Is that possible to cache StreamingResponse?

### DISCUSSION THREAD

**Discussion by mateusz-glowinski** [encode/starlette#1935] [Q&A]
Hi,
is there  a way to cache [StreamingResponse](https://www.starlette.io/responses/#streamingresponse)?

My initial idea was to convert `StreamingResponse` to `JSON`, but I couldn't find a nice way to do that.

**Comment by alex-oleshkevich** [comment]
Do you mean the generated response contents? If so then there is no difference between cached streaming response contents and usual JSON response that contains pregenerated data.

**Reply by alex-oleshkevich** [reply]
In this case you want to cache contents, not the response.

**Reply by mateusz-glowinski** [reply]
Yeap, I want to cache the content of the response, but there are some issues e.g. `Stream` can be infinite.
That is why I'm not really sure how to do that and I'm looking for already working solution.

**Reply by alex-oleshkevich** [reply]
Cache the result of your generator function (the one you pass into the streaming response). 
What is your task?

**Reply by mateusz-glowinski** [reply]
Yeap, but in that case i need to read the whole response. I'm not really convinced it's the way to go (although for now I have no other idea).

I've architecture like below:
```

                            =============== Docker Network =============
                            ||                                        ||
                            ||           -------> Backend Service A   ||
                            ||          /                             ||
                            ||         /                              ||
External Request ===> Gateway Service * --------> Backend Service B   ||
                            ||         \                              ||
                            ||          \                             ||
                            ||           -------> Backend Service C   ||
                            ||                                        ||
                            ============================================
```

The only way to access `Service A`, `Service B`, `Service C` is via `Gateway service`.
`Gateway service` passes responses to user from inner services via `StreamingResponse`.

My task is to cache responses from those services on `gateway service` for 5 sec.

**Reply by alex-oleshkevich** [reply]
Well, I am afraid there is no general solution for this. 
I would cache response contents if they are not larger than some limits (say 500kb) and stream the rest w/o caching.

**Reply by mateusz-glowinski** [reply]
ok, thank you for a tip :)
